MnemonicSwift/Mnemonic.swift

```diff
@@ -17,14 +17,45 @@ public enum MnemonicLanguageType {
     }
 }
 
+public enum MnemonicError: Error {
+    case wrongWordCount
+    case checksumError
+    case invalidWord(word: String)
+    case unsupportedLanguage
+    case invalidHexstring
+    case invalidBitString
+    case invalidInput
+    case entropyCreationFailed
+}
+
+public enum WordCount: Int {
+    case twelve = 12
+    case fifteen = 15
+    case eighteen = 18
+    case twentyOne = 21
+    case twentFour = 24
+
+    var bitLength: Int {
+        self.rawValue / 3 * 32
+    }
+
+    var checksumLength: Int {
+        self.rawValue / 3
+    }
+}
+
 public enum Mnemonic {
     /// Generate a mnemonic from the given hex string in the given language.
     ///
     /// - Parameters:
     ///   - hexString: The hex string to generate a mnemonic from.
     ///   - language: The language to use. Default is english.
-    public static func mnemonicString(from hexString: String, language: MnemonicLanguageType = .english) -> String? {
-        let seedData = hexString.mnemonicData()
+    /// - Returns: the mnemonic string or nil if input is invalid
+    /// - Throws:
+    ///   - `MnemonicError.InvalidHexString`:  when an invalid string is given
+    ///   - `MnemonicError.invalidBitString` when the resulting bitstring generates an invalid word index
+    static func mnemonicString(from hexString: String, language: MnemonicLanguageType = .english) throws -> String {
+        guard let seedData = hexString.mnemonicData() else { throw MnemonicError.invalidHexstring }
```

Comment:
Not a blocker but it would be nice to throw an exception here if `seedData` isn't 16, 20, 24, 28, or 32 bytes (corresponding to the valid word counts).

Reply:
you are right, also kind of completes BIP-39 specs definition into the code base